src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -107,15 +106,19 @@ public String generateQuery() {
                 queryMap.put(TO_DATE_FIELD, searchDateParams.getToDate());
             }
             if (isNotBlank(email)) {
-                queryMap.put(EMAIL_FIELD, LIKE_HOLDER.replace("LIKE", email));
+                queryMap.put(EMAIL_FIELD, likeClause(email));
             }
             if (isNotBlank(reference)) {
-                queryMap.put(REFERENCE_FIELD, LIKE_HOLDER.replace("LIKE", reference));
+                queryMap.put(REFERENCE_FIELD, likeClause(reference));
             }
             if (amount != null) {
                 queryMap.put(AMOUNT_FIELD, amount);
             }
         }
         return queryMap;
     }
+
+    private static String likeClause(String rawUserInputText) {
```

Comment:
are we planning to use it outside of this class? otherwise it shouldn't be static

Reply:
Sometimes I like to use `static` to indicate that a method is pure and won’t change anything in this instance of the class (I know `static` in Java doesn’t always guarantee this)

Reply:
IntelliJ is actually suggesting that I add 

```
    @org.jetbrains.annotations.NotNull
    @org.jetbrains.annotations.Contract(pure = true)
```

maybe we should start using these annotations?